pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -121,6 +121,15 @@ func (fr *fileReader) startWalkingDisk() error {
 
 func (fr *fileReader) deleteDashboardIfFileIsMissing(provisionedDashboardRefs map[string]*models.DashboardProvisioning, filesFoundOnDisk map[string]os.FileInfo) {
 	if fr.Cfg.DisableDeletion {
+		// If deletion is disabled for the provisioner we just remove provisioning metadata about the dashboard
+		// so afterwards the dashboard is considered unprovisioned.
+		for _, provisioningData := range provisionedDashboardRefs {
+			cmd := &models.UnprovisionDashboard{Id: provisioningData.DashboardId}
```

Comment:
I would like the `UnprovisionDashboard` to be implemented in the `DashboardProvisioningService`:
https://github.com/grafana/grafana/blob/6b8e3122f802cf94fa85ba0fb221b339434068a7/pkg/services/dashboards/dashboard_service.go#L20-L25

You probably still need to use the bus in that implementation and I noticed that you've defined `UnprovisionDashboard` in [dashboard models](https://github.com/grafana/grafana/pull/16211/files#diff-94a7531e86bbfcda538bde28d2ffb94bR360) - that should be named `UnprovisionDashboardCommand` and moved up to the other commands.

Naming is hard, don't know whether any of Unprovision, Deprovision or UnmarkDashboardAsProvisioned is better than the other. Thoughts?

Some tests of this would be nice, but may be tricky?

Reply:
Oh yeah I see now what you mean, as it is used in other functions in the `file_reader.go` will change that when I come from vacation.

Regarding not using the bus, in general, is there some long term plan? I hear someone mentioning that maybe it will go away. Or is the issue just that the indirect bus access should be used from inside of services and it will stay for that?

Regarding naming, no hard opinions on this one. Definitely a weird word `Unprovision` but probably makes most sense to me at the moment.

I definitely want to add some tests for this. I have seen that we have already a test DB setup, so I think it should not be too hard to create a more of an integration test by reading a test yaml file and checking the DB data based on that. Which I think would make sense as there is a bit more side effects and IO in this functionality to make a sensible unit test.

Reply:
> Regarding not using the bus, in general, is there some long term plan? I hear someone mentioning that maybe it will go away. Or is the issue just that the indirect bus access should be used from inside of services and it will stay for that?

Some want to remove it completely and some want to keep it. I personally think the bus has it's pros for some use cases. In some cases it's not optimal that the bus handlers are implemented in the sqlstore package. For this specific use case since provisioning are using provisioning service it's good to put the bus access in the service. Services is a rather new concept so we need to discuss this moving forward.